### ISSUE TITLE
Valid Virtual Service Definition Fails Config Validation

### ISSUE THREAD

**Issue by daaasbu** [kiali/kiali#764]
**Describe the bug**
It appears that Kiali is failing to validate a virtual service. I modified the virtual service from the tutorial for accessing google over https. It appears to be working from the istio side of things just fine but is failing the validation in the UI. I'm also not sure if this is the reason it is not showing the graph view as well.

https://istio.io/docs/tasks/traffic-management/egress/#make-requests-to-the-external-services

![virtual service error](https://user-images.githubusercontent.com/7543162/50521882-90f2ab80-0a8d-11e9-951e-878c734cab72.png)  
  
The error displayed is:  
*VirtualService doesn't define any route protocol.*  

**Versions used**
kiali-ui 0.6.0 (7ebeee3bec89b26235995d393ad99908b8fdb470)  
kiali v0.6.0 (23643f7f6a099041692205ba5affa7b3a91cabdd)  
Components  
Istio 1.0.0  
Prometheus 2.3.1  
Kubernetes v1.10.9-gke.5  

**To Reproduce**

Here is the virtual service yaml file. when using kubectl describe, it outputs what you would expect, but when displayed in the kiali UI it cuts off after host.   
  ```
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: avalara
spec:
  hosts:
  - sandbox-rest.avatax.com
  tls:
  - match:
    - port: 443
      sni_hosts:
      - sandbox-rest.avatax.com
    route:
    - destination:
        host: sandbox-rest.avatax.com
        port:
          number: 443
      weight: 100
```

**Expected behavior**
No validation error and the entire virtual service definition is shown.

**Comment by daaasbu** [comment]
So I have upgrade to: 
kiali-ui 0.10.1 (e58567cf961e281751b46582499c1e71ac619e97)  
kiali v0.10.1 (cc2251b69485e23793ba0b0bfa1c4198ede9653b)  
Components  
Istio 1.0.5  
Prometheus 2.3.1  
Kubernetes v1.10.9-gke.5    
And now it shows the full definition but now has the error:  
*Destination weight on route doesn't have a valid service*
![kiali](https://user-images.githubusercontent.com/7543162/50523307-001fce00-0a95-11e9-89bb-27185777b1af.png)

**Comment by abonas** [comment]
Hi @daaasbu, is there a DestinationRule that matches the destination section (specifically the host) as defined in this VirtualService?
Are you trying to access something outside of the mesh or inside the mesh?
@xeviknal  Is this particular validation message documented on Kiali website? I searched the validations tables and didn't spot this one - perhaps the message is slightly different?

**Comment by daaasbu** [comment]
@abonas is that an istio requirement or a kiali one? Because from the istio side of things everything is working just fine. Here is the guide I mentioned I followed:
https://istio.io/docs/tasks/traffic-management/egress/#make-requests-to-the-external-services

**Comment by israel-hdez** [comment]
@daaasbu do you have both the ServiceEntry and the VirtualService defined in the same namespace?

**Comment by pilhuhn** [comment]
I think Istio does not require a DR when there is only one possible destination in a VS, so that the implicit weight is 100%.
I think what @daaasbu is seeing may be a different incarnation of 
https://issues.jboss.org/browse/KIALI-2147

**Comment by daaasbu** [comment]
@pilhuhn yes, that could be it. I have mTLS disabled at the moment. And yes, they are both in the same namespace.

**Comment by burmanm** [comment]
This was fixed in PR #789, it's a result of not having any workloads that would trigger the serviceEntry lookup.
